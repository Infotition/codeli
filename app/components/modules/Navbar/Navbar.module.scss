.settings {
  opacity: 0;
  transform: translateY(-20px);
  transition: opacity 0.4s ease, transform 0.4s ease, visibility 0.4s;
  visibility: hidden;
}

.settings.active {
  opacity: 1;
  transform: translateY(0);
  visibility: visible;
}
