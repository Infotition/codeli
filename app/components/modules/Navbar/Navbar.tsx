import { useRef } from 'react';

import Brand from '@element/Brand/Brand';
import Icon, { Icons } from '@element/Icon/Icon';
import Tooltip, { TooltipDirection } from '@element/Tooltip/Tooltip';
import { ThemeProvider } from '@context/themeContext';
import useDetectOutsideClick from '@hook/useDetectOutsideClick';
import { useSidenav } from '@context/sidenavContext';
import MenuButton from '@element/MenuButton/MenuButton';
import SettingsMenu from '@element/SettingsMenu/SettingsMenu';

type NavItemProps = { children: React.ReactNode };

const NavItem = ({ children }: NavItemProps) => (
  <div className="hidden lg:flex lg:items-center lg:space-x-4 hover-icon">
    {children}
  </div>
);

const Navbar = () => {
  const settingsRef = useRef(null);
  const [settingsActive, setSettingsActive] = useDetectOutsideClick(
    settingsRef,
    false
  );

  const { dispatch, state } = useSidenav();

  return (
    <nav className="relative items-center w-full p-3 px-5 shadow-md bg1 text color-transition grid grid-cols-3">
      <div className="flex items-center justify-start space-x-12 lg:font-semibold lg:tracking-wider">
        <MenuButton
          className="hover-icon"
          height="20"
          width="30"
          onClick={() => dispatch({ type: 'toggle' })}
          isOpen={state.isActive}
        />

        <NavItem>
          <Icon icon={Icons.CHEVRON_LEFT} size={{ width: 22, height: 22 }} />
          <span>Vorherige</span>
        </NavItem>

        <NavItem>
          <span>NÃ¤chste</span>
          <Icon icon={Icons.CHEVRON_RIGHT} size={{ width: 22, height: 22 }} />
        </NavItem>
      </div>

      <Brand href={'/'} className="justify-center" />

      <div className="flex justify-end lg:space-x-5">
        <Tooltip tooltip="Diskussion" direction={TooltipDirection.BOTTOM_LEFT}>
          <Icon icon={Icons.CHAT} className="hidden lg:flex" />
        </Tooltip>
        <Tooltip
          tooltip="Einstellungen"
          direction={TooltipDirection.BOTTOM_LEFT}
        >
          <Icon
            icon={Icons.SETTINGS}
            onClick={() => setSettingsActive(!settingsActive)}
          />
        </Tooltip>
      </div>

      <ThemeProvider>
        <SettingsMenu toggleRev={settingsRef} isActive={settingsActive} />
      </ThemeProvider>
    </nav>
  );
};

export default Navbar;
