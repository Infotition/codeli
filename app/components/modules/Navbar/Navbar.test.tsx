import { render } from '@testing-library/react';
import Navbar from './Navbar';

describe('Module Header', () => {
  it('should render component', () => {
    const header = render(<Navbar />);
    expect(header).toMatchSnapshot();
  });
});
