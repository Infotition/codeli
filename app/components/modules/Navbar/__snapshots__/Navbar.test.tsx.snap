// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module Header should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class=""
  >
    <div>
      <nav
        class="relative items-center p-3 shadow-md bg1 text color-transition grid grid-cols-3"
      >
        <div
          class="flex justify-start space-x-12 lg:font-semibold lg:tracking-wider"
        >
          <svg
            class="undefined hover-icon select-none"
            fill="currentColor"
            focusable="false"
            height="30"
            stroke="currentColor"
            viewBox="0 0 24 24"
            width="30"
          >
            burger icon
            <path
              d="M4 6h16M4 12h16M4 18h16"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <div
            class="hidden lg:flex lg:items-center lg:space-x-4 hover-icon"
          >
            <svg
              class="undefined hover-icon select-none"
              fill="currentColor"
              focusable="false"
              height="22"
              stroke="currentColor"
              viewBox="0 0 24 24"
              width="22"
            >
              chevron_left icon
              <path
                d="M15 19l-7-7 7-7"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <span>
              Vorherige
            </span>
          </div>
          <div
            class="hidden lg:flex lg:items-center lg:space-x-4 hover-icon"
          >
            <span>
              NÃ¤chste
            </span>
            <svg
              class="undefined hover-icon select-none"
              fill="currentColor"
              focusable="false"
              height="22"
              stroke="currentColor"
              viewBox="0 0 24 24"
              width="22"
            >
              chrevron_right icon
              <path
                d="M9 5l7 7-7 7"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
        </div>
        <a
          class="justify-center text-primary flex sm:items-center"
          href="/"
        >
          <svg
            class="undefined false select-none"
            fill="currentColor"
            focusable="false"
            height="50"
            stroke="currentColor"
            viewBox="0 0 24 24"
            width="50"
          >
            infotition icon
            <g
              transform="translate(-19 -15) scale(0.27)"
            >
              <path
                d="m95.188 101.22c0 0.94809-0.43546 1.8342-1.1644 2.4405-0.98943 0.82407-1.6192 2.0643-1.6192 3.452 0 1.8769 1.1507 3.4851 2.7836 4.1562 0.52503 0.21359 1.0983 0.33073 1.7005 0.33073 0.60358 0 1.1768-0.11714 1.7019-0.33073 1.633-0.67111 2.7836-2.2793 2.7836-4.1562 0-1.3877-0.62976-2.6279-1.6192-3.452-0.7276-0.60634-1.1644-1.4924-1.1644-2.4405v-37.735h-6.9563c-0.93844 0-1.7005 0.76068-1.7005 1.7005s0.76205 1.7005 1.7005 1.7005h3.554z"
              />
              <path
                d="m89.934 58.38c0 0.93982 0.76205 1.7005 1.7005 1.7005h39.667c0.93844 0 1.7005-0.76068 1.7005-1.7005s-0.76206-1.7005-1.7005-1.7005h-39.667c-0.93844 0-1.7005 0.76068-1.7005 1.7005"
              />
              <path
                d="m91.634 142.21h39.667c0.93844 0 1.7005-0.76067 1.7005-1.7005 0-0.93983-0.76206-1.7019-1.7005-1.7019h-39.667c-0.93844 0-1.7005 0.76205-1.7005 1.7019 0 0.93982 0.76205 1.7005 1.7005 1.7005"
              />
              <path
                d="m98.591 135.4h9.7193v-48.872c0-1.1203 0.42581-2.2269 1.265-2.9669 0.93155-0.82131 1.5186-2.0243 1.5186-3.3638 0-1.8769-1.152-3.4892-2.7836-4.1562-0.52504-0.21773-1.0983-0.33486-1.7019-0.33486-0.6022 0-1.1755 0.11713-1.7005 0.33486-1.633 0.66697-2.785 2.2793-2.785 4.1562 0 1.3395 0.58842 2.5425 1.5186 3.3638 0.84061 0.74001 1.2664 1.8466 1.2664 2.9669v45.471h-6.3169v-15.322c0-0.92604-0.75792-1.6826-1.684-1.6812h-0.03443c-0.92604-1e-3 -1.684 0.75516-1.684 1.6812v15.322h-3.554c-0.93845 0-1.7005 0.76068-1.7005 1.7005s0.76205 1.7005 1.7005 1.7005z"
              />
              <path
                d="m118.03 63.483h-13.12v7.192c0 0.92191 0.76068 1.6399 1.6826 1.6371h0.0358c0.92192 0.0028 1.6826-0.7152 1.6826-1.6371v-3.791h6.3183v48.309c0 1.1217-0.42582 2.2283-1.2664 2.971-0.93018 0.82269-1.5186 2.0257-1.5186 3.3652 0 1.8769 1.152 3.4851 2.785 4.1562 0.52503 0.21222 1.0983 0.32935 1.7005 0.32935 0.60221 0 1.1755-0.11713 1.7005-0.32935 1.633-0.6711 2.785-2.2793 2.785-4.1562 0-1.3394-0.58842-2.5425-1.5186-3.3652-0.8406-0.74276-1.2664-1.8493-1.2664-2.971z"
              />
              <path
                d="m124.35 63.483v27.982c0 0.50574-0.23978 0.9977-0.66973 1.265-1.2706 0.79237-2.1153 2.2007-2.1153 3.8075s0.84474 3.0152 2.1153 3.8075c0.42995 0.26734 0.66973 0.75792 0.66973 1.265v30.391h-6.3183v-0.94809c0-0.90537-0.73449-1.6385-1.6399-1.6371h-0.12127c-0.90537-1e-3 -1.6399 0.73174-1.6399 1.6371v4.3491h16.674c0.93982 0 1.7005-0.76206 1.7005-1.7005 0-0.93982-0.76068-1.7005-1.7005-1.7005h-3.554v-30.391c0-0.50712 0.2384-0.9977 0.66834-1.265 1.2706-0.79237 2.1167-2.2007 2.1167-3.8075s-0.84612-3.0152-2.1167-3.8075c-0.42994-0.26734-0.66834-0.7593-0.66834-1.265v-24.581h3.554c0.93982 0 1.7005-0.76068 1.7005-1.7005s-0.76068-1.7005-1.7005-1.7005z"
              />
            </g>
          </svg>
          <span
            class="hidden select-none sm:block sm:font-semibold sm:text-5xl sm:-m-4"
          >
            NFOTITION
          </span>
        </a>
        <div
          class="flex justify-end lg:space-x-5"
        >
          <div
            class="relative flex flex-row-reverse group"
          >
            <svg
              class="hidden lg:flex hover-icon select-none"
              fill="currentColor"
              focusable="false"
              height="25"
              stroke="currentColor"
              viewBox="0 0 24 24"
              width="25"
            >
              chat icon
              <path
                d="M8 10h.01M12 10h.01M16 10h.01M9 16H5a2 2 0 01-2-2V6a2 2 0 012-2h14a2 2 0 012 2v8a2 2 0 01-2 2h-5l-5 5v-5z"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <span
              class="origin-top-right top-[130%] top-right-arrow absolute px-10 py-3 text-lg font-semibold shadow-md text bg3 rounded-md transition-all duration-300 scale-0 lg:group-hover:scale-100"
              style="margin-right: 0px;"
            >
              Diskussion
            </span>
          </div>
          <div
            class="relative flex flex-row-reverse group"
          >
            <svg
              class="undefined hover-icon select-none"
              fill="currentColor"
              focusable="false"
              height="25"
              stroke="currentColor"
              viewBox="0 0 24 24"
              width="25"
            >
              settings icon
              <g
                fill="none"
              >
                <path
                  d="M10.325 4.317c.426-1.756 2.924-1.756 3.35 0a1.724 1.724 0 002.573 1.066c1.543-.94 3.31.826 2.37 2.37a1.724 1.724 0 001.065 2.572c1.756.426 1.756 2.924 0 3.35a1.724 1.724 0 00-1.066 2.573c.94 1.543-.826 3.31-2.37 2.37a1.724 1.724 0 00-2.572 1.065c-.426 1.756-2.924 1.756-3.35 0a1.724 1.724 0 00-2.573-1.066c-1.543.94-3.31-.826-2.37-2.37a1.724 1.724 0 00-1.065-2.572c-1.756-.426-1.756-2.924 0-3.35a1.724 1.724 0 001.066-2.573c-.94-1.543.826-3.31 2.37-2.37.996.608 2.296.07 2.572-1.065z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
                <path
                  d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </g>
            </svg>
            <span
              class="origin-top-right top-[130%] top-right-arrow absolute px-10 py-3 text-lg font-semibold shadow-md text bg3 rounded-md transition-all duration-300 scale-0 lg:group-hover:scale-100"
              style="margin-right: 0px;"
            >
              Einstellungen
            </span>
          </div>
        </div>
        <div
          class="settings false absolute ml-8 top-32 p-5 right-5 shadow-lg justify-end color-transition"
        >
          <div
            class="relative flex flex-row-reverse group"
          >
            <div
              style="opacity: 1; transform: none;"
            >
              <svg
                class="undefined hover-icon select-none"
                fill="currentColor"
                focusable="false"
                height="25"
                stroke="currentColor"
                viewBox="0 0 24 24"
                width="25"
              >
                sun icon
                <path
                  d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z"
                  fill="none"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
            <span
              class="origin-right right-arrow absolute px-10 py-3 text-lg font-semibold shadow-md text bg3 rounded-md transition-all duration-300 scale-0 lg:group-hover:scale-100"
              style="margin-right: 0px;"
            >
              Theme
            </span>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="relative items-center p-3 shadow-md bg1 text color-transition grid grid-cols-3"
    >
      <div
        class="flex justify-start space-x-12 lg:font-semibold lg:tracking-wider"
      >
        <svg
          class="undefined hover-icon select-none"
          fill="currentColor"
          focusable="false"
          height="30"
          stroke="currentColor"
          viewBox="0 0 24 24"
          width="30"
        >
          burger icon
          <path
            d="M4 6h16M4 12h16M4 18h16"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <div
          class="hidden lg:flex lg:items-center lg:space-x-4 hover-icon"
        >
          <svg
            class="undefined hover-icon select-none"
            fill="currentColor"
            focusable="false"
            height="22"
            stroke="currentColor"
            viewBox="0 0 24 24"
            width="22"
          >
            chevron_left icon
            <path
              d="M15 19l-7-7 7-7"
              fill="none"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <span>
            Vorherige
          </span>
        </div>
        <div
          class="hidden lg:flex lg:items-center lg:space-x-4 hover-icon"
        >
          <span>
            NÃ¤chste
          </span>
          <svg
            class="undefined hover-icon select-none"
            fill="currentColor"
            focusable="false"
            height="22"
            stroke="currentColor"
            viewBox="0 0 24 24"
            width="22"
          >
            chrevron_right icon
            <path
              d="M9 5l7 7-7 7"
              fill="none"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
      </div>
      <a
        class="justify-center text-primary flex sm:items-center"
        href="/"
      >
        <svg
          class="undefined false select-none"
          fill="currentColor"
          focusable="false"
          height="50"
          stroke="currentColor"
          viewBox="0 0 24 24"
          width="50"
        >
          infotition icon
          <g
            transform="translate(-19 -15) scale(0.27)"
          >
            <path
              d="m95.188 101.22c0 0.94809-0.43546 1.8342-1.1644 2.4405-0.98943 0.82407-1.6192 2.0643-1.6192 3.452 0 1.8769 1.1507 3.4851 2.7836 4.1562 0.52503 0.21359 1.0983 0.33073 1.7005 0.33073 0.60358 0 1.1768-0.11714 1.7019-0.33073 1.633-0.67111 2.7836-2.2793 2.7836-4.1562 0-1.3877-0.62976-2.6279-1.6192-3.452-0.7276-0.60634-1.1644-1.4924-1.1644-2.4405v-37.735h-6.9563c-0.93844 0-1.7005 0.76068-1.7005 1.7005s0.76205 1.7005 1.7005 1.7005h3.554z"
            />
            <path
              d="m89.934 58.38c0 0.93982 0.76205 1.7005 1.7005 1.7005h39.667c0.93844 0 1.7005-0.76068 1.7005-1.7005s-0.76206-1.7005-1.7005-1.7005h-39.667c-0.93844 0-1.7005 0.76068-1.7005 1.7005"
            />
            <path
              d="m91.634 142.21h39.667c0.93844 0 1.7005-0.76067 1.7005-1.7005 0-0.93983-0.76206-1.7019-1.7005-1.7019h-39.667c-0.93844 0-1.7005 0.76205-1.7005 1.7019 0 0.93982 0.76205 1.7005 1.7005 1.7005"
            />
            <path
              d="m98.591 135.4h9.7193v-48.872c0-1.1203 0.42581-2.2269 1.265-2.9669 0.93155-0.82131 1.5186-2.0243 1.5186-3.3638 0-1.8769-1.152-3.4892-2.7836-4.1562-0.52504-0.21773-1.0983-0.33486-1.7019-0.33486-0.6022 0-1.1755 0.11713-1.7005 0.33486-1.633 0.66697-2.785 2.2793-2.785 4.1562 0 1.3395 0.58842 2.5425 1.5186 3.3638 0.84061 0.74001 1.2664 1.8466 1.2664 2.9669v45.471h-6.3169v-15.322c0-0.92604-0.75792-1.6826-1.684-1.6812h-0.03443c-0.92604-1e-3 -1.684 0.75516-1.684 1.6812v15.322h-3.554c-0.93845 0-1.7005 0.76068-1.7005 1.7005s0.76205 1.7005 1.7005 1.7005z"
            />
            <path
              d="m118.03 63.483h-13.12v7.192c0 0.92191 0.76068 1.6399 1.6826 1.6371h0.0358c0.92192 0.0028 1.6826-0.7152 1.6826-1.6371v-3.791h6.3183v48.309c0 1.1217-0.42582 2.2283-1.2664 2.971-0.93018 0.82269-1.5186 2.0257-1.5186 3.3652 0 1.8769 1.152 3.4851 2.785 4.1562 0.52503 0.21222 1.0983 0.32935 1.7005 0.32935 0.60221 0 1.1755-0.11713 1.7005-0.32935 1.633-0.6711 2.785-2.2793 2.785-4.1562 0-1.3394-0.58842-2.5425-1.5186-3.3652-0.8406-0.74276-1.2664-1.8493-1.2664-2.971z"
            />
            <path
              d="m124.35 63.483v27.982c0 0.50574-0.23978 0.9977-0.66973 1.265-1.2706 0.79237-2.1153 2.2007-2.1153 3.8075s0.84474 3.0152 2.1153 3.8075c0.42995 0.26734 0.66973 0.75792 0.66973 1.265v30.391h-6.3183v-0.94809c0-0.90537-0.73449-1.6385-1.6399-1.6371h-0.12127c-0.90537-1e-3 -1.6399 0.73174-1.6399 1.6371v4.3491h16.674c0.93982 0 1.7005-0.76206 1.7005-1.7005 0-0.93982-0.76068-1.7005-1.7005-1.7005h-3.554v-30.391c0-0.50712 0.2384-0.9977 0.66834-1.265 1.2706-0.79237 2.1167-2.2007 2.1167-3.8075s-0.84612-3.0152-2.1167-3.8075c-0.42994-0.26734-0.66834-0.7593-0.66834-1.265v-24.581h3.554c0.93982 0 1.7005-0.76068 1.7005-1.7005s-0.76068-1.7005-1.7005-1.7005z"
            />
          </g>
        </svg>
        <span
          class="hidden select-none sm:block sm:font-semibold sm:text-5xl sm:-m-4"
        >
          NFOTITION
        </span>
      </a>
      <div
        class="flex justify-end lg:space-x-5"
      >
        <div
          class="relative flex flex-row-reverse group"
        >
          <svg
            class="hidden lg:flex hover-icon select-none"
            fill="currentColor"
            focusable="false"
            height="25"
            stroke="currentColor"
            viewBox="0 0 24 24"
            width="25"
          >
            chat icon
            <path
              d="M8 10h.01M12 10h.01M16 10h.01M9 16H5a2 2 0 01-2-2V6a2 2 0 012-2h14a2 2 0 012 2v8a2 2 0 01-2 2h-5l-5 5v-5z"
              fill="none"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <span
            class="origin-top-right top-[130%] top-right-arrow absolute px-10 py-3 text-lg font-semibold shadow-md text bg3 rounded-md transition-all duration-300 scale-0 lg:group-hover:scale-100"
            style="margin-right: 0px;"
          >
            Diskussion
          </span>
        </div>
        <div
          class="relative flex flex-row-reverse group"
        >
          <svg
            class="undefined hover-icon select-none"
            fill="currentColor"
            focusable="false"
            height="25"
            stroke="currentColor"
            viewBox="0 0 24 24"
            width="25"
          >
            settings icon
            <g
              fill="none"
            >
              <path
                d="M10.325 4.317c.426-1.756 2.924-1.756 3.35 0a1.724 1.724 0 002.573 1.066c1.543-.94 3.31.826 2.37 2.37a1.724 1.724 0 001.065 2.572c1.756.426 1.756 2.924 0 3.35a1.724 1.724 0 00-1.066 2.573c.94 1.543-.826 3.31-2.37 2.37a1.724 1.724 0 00-2.572 1.065c-.426 1.756-2.924 1.756-3.35 0a1.724 1.724 0 00-2.573-1.066c-1.543.94-3.31-.826-2.37-2.37a1.724 1.724 0 00-1.065-2.572c-1.756-.426-1.756-2.924 0-3.35a1.724 1.724 0 001.066-2.573c-.94-1.543.826-3.31 2.37-2.37.996.608 2.296.07 2.572-1.065z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </g>
          </svg>
          <span
            class="origin-top-right top-[130%] top-right-arrow absolute px-10 py-3 text-lg font-semibold shadow-md text bg3 rounded-md transition-all duration-300 scale-0 lg:group-hover:scale-100"
            style="margin-right: 0px;"
          >
            Einstellungen
          </span>
        </div>
      </div>
      <div
        class="settings false absolute ml-8 top-32 p-5 right-5 shadow-lg justify-end color-transition"
      >
        <div
          class="relative flex flex-row-reverse group"
        >
          <div
            style="opacity: 1; transform: none;"
          >
            <svg
              class="undefined hover-icon select-none"
              fill="currentColor"
              focusable="false"
              height="25"
              stroke="currentColor"
              viewBox="0 0 24 24"
              width="25"
            >
              sun icon
              <path
                d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
          <span
            class="origin-right right-arrow absolute px-10 py-3 text-lg font-semibold shadow-md text bg3 rounded-md transition-all duration-300 scale-0 lg:group-hover:scale-100"
            style="margin-right: 0px;"
          >
            Theme
          </span>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
